### PR TITLE
Doc fetching

### PR DESCRIPTION
**Pull request by DertelyanMonika**
Added fetching functionality

### CODE REVIEW

File: pycognaize/document/document.py [hunk 1]
```diff
@@ -26,6 +31,46 @@
 from pycognaize.document.tag.tag import BoxTag, LineTag
 
 
+class ApiConfig(Enum):
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: Move this to enums

File: pycognaize/document/document.py [hunk 2]
```diff
@@ -26,6 +31,46 @@
...
+    API_PREFIX = '/api/v1/integration/'
+    CREATE_TASK_ENDPOINT = f"{API_PREFIX}document/modelinput"
+    RUN_MODEL_ENDPOINT = f"{API_PREFIX}data"
+    DEFAULT_TIMEOUT = 600
+    RETRIES = Retry(total=3,
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: Remove this from enum

Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: Rename to RETRY_ADAPTER

File: pycognaize/document/document.py [hunk 3]
```diff
@@ -26,6 +31,46 @@
...
+                    backoff_factor=10,
+                    status_forcelist=[500, 502, 503, 504])
+
+
+def create_and_get_task_id(document_id: str, recipe_id: str) -> str:
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: make private

File: pycognaize/document/document.py [hunk 4]
```diff
@@ -26,6 +31,46 @@
...
+     id and recipe id"""
+    url = os.environ.get('API_HOST') + ApiConfig.CREATE_TASK_ENDPOINT.value
+    payload = "{\"documentId\": \"%s\",\n \"recipeId\": \"%s\"\n}\n" % (
+        document_id, recipe_id)
+    headers = {'x-auth': os.environ.get('X_AUTH_TOKEN'), 'content-type': "application/json"}
+    return requests.request("POST", url, data=payload,
+                            headers=headers).json()['taskId']
+
+
+def get_document_by_task_id(task_id: str,
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: make private

File: pycognaize/document/document.py [hunk 5]
```diff
@@ -89,6 +134,14 @@ def html(self):
         """Returns `HTML` object"""
         return self._html_info
 
+    @classmethod
+    def fetch_document(cls, recipe_id, doc_id):
+        """Get the document object, given a document id and recipe id"""
+        task_id = create_and_get_task_id(document_id=doc_id,
+                                         recipe_id=recipe_id)
+        doc = get_document_by_task_id(task_id=task_id)
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: document

File: pycognaize/document/document.py [hunk 6]
```diff
@@ -89,6 +134,14 @@ def html(self):
...
+        return doc
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: doc: Document = Document.from_dict(document_json,
                                       data_path=doc_data_path)

File: pycognaize/document/document.py (hunk 5 above)
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: doc_data_path, document_json = ....

File: pycognaize/document/document.py [hunk 7]
```diff
@@ -563,4 +616,4 @@ def annotate_pdf(doc: fitz.Document,
     annot.set_colors(color_dict)
     annot.set_opacity(opacity)
     annot.update()
-    return doc.write()
+    return doc.write()
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: Add empty line please

File: pycognaize/document/document.py [hunk 8]
```diff
@@ -58,6 +63,39 @@ def y(self) -> 'FieldCollection[str, List[Field]]':
...
+        get_response: requests.Response = \
+            session.get(os.environ.get('API_HOST') +
+                        ApiConfigEnum.RUN_MODEL_ENDPOINT.value +
+                        '/' + task_id, verify=False,
+                        timeout=ApiConfigEnum.DEFAULT_TIMEOUT.value)
+        get_response.raise_for_status()
+        get_response_dict: dict = get_response.json()
+        doc_data_path: str = get_response_dict['documentRootPath']
+        document_json: dict = get_response_dict['inputDocument']
+        # doc: Document = Document.from_dict(document_json,
```
Comment (hovhannes-zohrabyan) [inline pycognaize/document/document.py]: delete comment

**Review by davidcognaize — APPROVED** [review]

**Pull request merged**